### PR TITLE
feat(tag component): implementation of tag and filter tag components

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**Tag (Gene)**
Shortcut ticket: [sh-160284](https://app.shortcut.com/sci-design-system/story/160284/implement-tags)

Tags provide a simplified API surface by wrapping the Material UI [Chip](https://material-ui.com/components/chips/) component and exposing a subset of the available props.

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/Tag/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    | "beta"
+    | [string, string]
+    | [string, string, string];
+}
+
+type TagProps = SdsTagProps;
+
+export type { TagProps };
+
+const Tag = (props: SdsTagProps): JSX.Element => {
```
Comment (tihuan) [inline src/core/Tag/index.tsx:26]: Can we add MUI's component link here as a comment please? This way it's easier for people to find the underlying MUI component!

(This is a new convention SDS wants to add to all applicable components 👌 ) Thanks so much!

e.g.,

```tsx
/**
 * @see: https://v4.mui.com/components/chips/
 */
const Tag = (props: SdsTagProps): JSX.Element => {
```

**Review by tihuan — COMMENTED** [review]
Thanks so much again for adding this component, Masoud 🎉 !! Will do a full review later today 🙏

**Review by tihuan — APPROVED** [review]
Code and storybook look super good! Thanks so much for the awesome PR, @masoudmanson 🤩 🙏 🚀 !

We just need a designer to check storybook again to approve styles

Otherwise LGTM!

File: src/core/Tag/index.tsx (hunk 1 above)
Comment (masoudmanson) [reply src/core/Tag/index.tsx:26]: Excellent points. I'll update the branch accordingly :D

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
See comments in chromatic

**Comment by masoudmanson** [comment]
@chanzuckerberg/sds-design 
Could you please review the Tags component?

**Review by clarsen-czi — APPROVED** [review]
LGTM!! Thanks for helping me sort out which PR to take a look at, @masoudmanson !

**Pull request merged**